tests/enrol_advanced_testcase_test.php

```diff
@@ -0,0 +1,330 @@
+     $this->enrolments = $DB->count_records('user_enrolments', array('enrolid'=>$this->user_enrolment->id));
+  //   var_dump($this->user_enrolment->id);
+   }
+
+   /*Simuation test if plugin is enabled*/
+   /**
+   * @test
+   */
+   public function basic(){
+     $anlass = $this->simulator->get_event_by_number("mod.mmpAUKATE1.HS18_BS.002");
+     $personen_anmeldung = $this->simulator->get_enrolments_by_eventid(25490);
+     //var_dump($personen_anmeldung);
+     $personen_anmeldung = $this->simulator->get_enrolments_by_eventid(25490);
+
+     //var_dump($this->simulator->evento_personen_anmeldungen);
+
+     $person = $this->simulator->get_person_by_id(141703);
+     //var_dump($person);
+     $person = $this->simulator->get_person_by_id(117828);
+     //var_dump($person);
+     $ad_account = $this->simulator->get_ad_accounts_by_evento_personid(141701, null, null);
+     //var_dump($ad_account);
+     $ad_account_student = $this->simulator->get_all_ad_accounts(null);
+     //var_dump($ad_account_student);
+}
+
+   /*Basic test if plugin is enabled*/
+   /**
+   * @test
+   */
+   public function basics(){
+
+     $this->resetAfterTest(true);
+
+     $this->enable_plugin();
+     $plugin = 'evento';
+     $evento_plugin = enrol_get_plugin($plugin);
+
+     $this->assertEquals( $evento_plugin->get_name(), 'evento');
+     $this->assertNotEmpty( $evento_plugin);
+   }
+   /*get_user() Test for a new user*/
+   /**
+   * @test
+   */
+   public function get_user(){
+     /*set global DB variable*/
+     global $DB;
+
+     /*Get Plugin name*/
+     $evento = new enrol_evento\task\evento_member_sync_task();
+     $name = $evento->get_name();
+
+     /*Get new user*/
+     $eventoperson = $this->locallib->get_user_exposed(141703, $isstudent=true, $username=null);
+     $eventoperson = $this->locallib->get_user_exposed(117828, $isstudent=true, $username=null);
+
+     /*Get Database Records*/
+     $user1 = $DB->get_record('user', array('username'=>'2460181390-1097805571-3701207438-51315@fh-htwchur.ch'));
+     $user2 = $DB->get_record('user', array('username'=>'2460181391-1097805571-3701207438-51315@fh-htwchur.ch'));
+
+     /*^Database Record equals new user*/
+     $this->assertEquals($name, 'Evento synchronisation');
+     $this->assertEquals($user1->email, 'hans.meier@stud.htwchur.ch');
+     $this->assertEquals($user2->email, 'max.muster@stud.htwchur.ch');
+   }
+
+   /**
+   * @test
+   */
+   public function get_ad_user(){
+     /*set evento person ID*/
+     $eventopersonid = 141703;
+     /*Get ad User*/
+     $person = $this->locallib->get_ad_user_exposed($eventopersonid, $isstudent=null);
+     /*Accountname  equals ad username*/
+     $this->assertEquals(current($person)->sAMAccountName, 'HanMei');
+   }
+
+   /*Get user by evento id test*/
+   /**
+   * @test
+   */
+   public function get_users_by_eventoid(){
+     /*Set the evento person ID*/
+     $this->get_user();
```

Comment:
`$this->get_user()` ist eine Testfunktion (hat das Attribut @test gesetzt.)
Das ist nicht üblich, dass man eine Testfunktion in einer anderen Testfunktion aufruft.